spec/component/mailers/deputy_assignments_mailer_spec.rb

```diff
@@ -3,25 +3,31 @@
 require 'component/component_spec_helper'
 
 describe DeputyAssignmentsMailer, type: :model do
-  let(:primary) { double 'user',
-                         email: 'primary@psu.edu',
-                         name: 'Primary User',
-                         webaccess_id: 'abc123' }
+  include Rails.application.routes.url_helpers
 
-  let(:deputy) { double 'user',
-                        email: 'deputy@psu.edu',
-                        name: 'Deputy User',
-                        webaccess_id: 'def456' }
+  let(:primary) { instance_double 'User',
```

Comment:
Just curious - what is the benefit here of using `instance_double` over `double`?

Reply:
`instance_double` ensures that you can only mock real actual methods, whereas `double` allows you to mock any method you want, regardless of whether it exists.

So for example in this very code the `User` model does not actually have a method/field called `email`. When using `double`, it allows this, which can lead to bugs. When I changed it over to `instance_double`, the tests exploded saying "you can't set `email` on `User` because User doesn't have one of those!"

So, tldr, always use `instance_double` or `instance_spy` if you can.